notifs/relredisnotif.go

```diff
@@ -0,0 +1,28 @@
+package notifs
+
+type NotificationType int
+
+const (
+	StreamAdded NotificationType = iota
+	StreamDisowned
+	StreamExpired
+	General
+)
+
+type RecoverableRedisNotification[T any] struct {
+	Type         NotificationType
+	Notification T
+}
```

Comment:
I am thinking it should be fine even if there is no `Type` field?

With the type, essentially the caller has to do the follwing: 
```go
switch Type {
    case X:
        // Cast Notification to X's concrete type
}
```

They may as well do:
```go
switch Notification.(type) {
    case X:
        // cast to X
}
```

Reply:
You're right and this is what I thought earlier too. I just thought:
1. An explicit typed field is better than clients figuring out based on the struct.
2. There need not be 1:1 mapping between structs and types. Two types can have one struct or may be of type string.